corehq/apps/userreports/exceptions.py

```diff
@@ -64,3 +67,15 @@ class InvalidQueryColumn(UserReportsError):
 
 class InvalidDataSourceType(UserQueryError):
     pass
+
+
+def translate_programming_error(exception):
+    if isinstance(exception, ProgrammingError):
+        orig = getattr(exception, 'orig')
+        if orig:
+            error_code = getattr(orig, 'pgcode')
+            # http://www.postgresql.org/docs/9.4/static/errcodes-appendix.html
+            if error_code == '42P01':
+                return TableNotFoundWarning
+            elif error_code == '42703':
+                return MissingColumnWarning
```

Comment:
added the exception message instead of all the details: fb3ba7b7aa. I considered using `six.reraise` or `six.raise_from` but since the one case only wants to raise the `TableNotFoundWarning` that doesn't work so well.